cognite/client/_api/data_modeling/graphql.py

```diff
@@ -133,3 +133,29 @@ def apply_dml(
         query_name = "upsertGraphQlDmlVersion"
         res = self._post_graphql(url_path="/dml/graphql", query_name=query_name, json=payload)
         return DMLApplyResult.load(res[query_name]["result"])
+
+    def query(self, id: DataModelIdentifier, query: str) -> dict[str, Any]:
+        """Apply the DML for a given data model.
+
+        Args:
+            id (DataModelIdentifier): The data model to apply DML to.
+            query (str): The query to issue.
+
+        Returns:
+            dict[str, Any]: The query result
+
+        Examples:
+
+            Execute a graphql query against a given data model::
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> res = c.data_modeling.graphql.query(
+                ...     id=("mySpace", "myDataModel", "v1"),
+                ...     query="listThings { items { thingProperty } }",
+                ... )
+        """
+        dm_id = DataModelId.load(id)
+        endpoint = f"/userapis/spaces/{dm_id.space}/datamodels/{dm_id.external_id}/versions/{dm_id.version}/graphql"
```

Comment:
Should we need to url encode this?

Reply:
Absolutely. Fixed. I wrongly assumed APIClient/HttpClient would do that for us. (We should probably make it so).